### PR TITLE
Improvements to Debugger GUI

### PR DESCRIPTION
**Pull request by hediet**
- [x] generalize from Methods to arbitrary Graphs
- [x] filters runtime functions by default
- [x] adds ENV variables `FILTER_BREAKPOINT_LABEL=[label,]*`, `FILTER_BREAKPOINT_LOCATION=[file:line,]*`, `FILTER_GRAPH_NAME=[name,]*`
- [x] add REST endpoints  to add filters
- [ ] ~add GUI buttons to filter graphs and breakpoints~ Might be implemented in a future PR.

### CODE REVIEW

File: compiler-lib/src/debugging.rs [hunk 1]
```diff
@@ -223,6 +226,31 @@ impl Debugger {
 
 struct DebuggerState(Debugger);
 
+pub struct CORS();
```
Comment (flip1995) [inline compiler-lib/src/debugging.rs]: NIT:
```suggestion
pub struct CORS;
```

File: compiler-lib/src/optimization/constant_folding.rs [hunk 2]
```diff
@@ -147,6 +147,27 @@ impl ConstantFolding {
...
+            }
+
+            if node == cur_node {
+                label = label
+                    .style(Style::Filled)
+                    .fillcolor(X11Color::Blue)
+                    .fontcolor(X11Color::White);
+            }
+
+            if let Some(priority) = self.queue.get(&node) {
```
Comment (flip1995) [inline compiler-lib/src/optimization/constant_folding.rs]:
```suggestion
            if let Some(_) = self.queue.get(&node) {
```

**Comment by hediet** [comment]
@reiner-dolp Fixed, even though I considered these issues irrelevant as it is somewhat hacky anyways.

**Pull request merged**